### PR TITLE
CA-85 Added post and delete routes to the FAQ router

### PR DESCRIPTION
**Pull request by spiray**
I added routes to the FAQ router for adding and deleting FAQs. Will we need a put/patch as well for modifying FAQs?

### CODE REVIEW

**Comment by spiray** [comment]
[As an Admin, I would like to be able to add and delete FAQs](https://trello.com/c/4ZjPegh2/85-as-an-admin-i-would-like-to-be-able-to-add-and-delete-faqs)

File: src/api/faq/faq.controller.js
Comment (ewenjo) [inline src/api/faq/faq.controller.js]: Should this include a status code in the response?
`{ message: 'FAQ added successfully to DB.', status: 200 }`

File: src/api/faq/faq.controller.js
Comment (ewenjo) [inline src/api/faq/faq.controller.js]: Error code 400 would fit better, since it is an invalid/bad request.
`const status = error instanceof ReferenceError ? 400 : 500;`

File: src/api/faq/faq.controller.js
Comment (ewenjo) [inline src/api/faq/faq.controller.js]: This is only called if the deletion above was successful, if the promise failes, it is caught by the trycatch.

It also happens if you try to fetch a FAQ with a ID that doens't exist, but that isn't a part of this PR.

File: src/api/faq/faq.controller.js
Comment (ewenjo) [inline src/api/faq/faq.controller.js]: Return with status code?

**Review by ewenjo — COMMENTED** [review]
Insertion and deletion works fine, but there are a few other things. See comments.

File: src/api/faq/faq.controller.js
Comment (spiray) [reply src/api/faq/faq.controller.js]: @ewenjo The way mongoose is behaving seems to handle all cases. It could be I am missing something here. When I was testing it, if an invalid ID was requested it wouldn't throw, it would just be successful with a deleted count of 0.

Comment (ewenjo) [reply src/api/faq/faq.controller.js]: It turns out it only throws an error if you try to delete a FAQ with an invalid id, Mongodb validates an id before it tries to use it. Something like "123" or "abc" will be invalid ids and give the output `{"status":500,"message":"Cast to ObjectId failed for value \"123\" at path \"_id\" for model \"faqs\""}`. 

But if I delete a FAQ and then try to delete it again, I get the proper error message. It's probably not necessary to cover these cases with a custom error message.

Comment (ewenjo) [reply src/api/faq/faq.controller.js]: Anything with 12 characters or more is a valid MongoDB object ID.
`if (_id.length < 12) throw new ReferenceError('FAQ ID must be at least 12 characters..');`

Comment (spiray) [reply src/api/faq/faq.controller.js]: So you think i should make sure the length is correct and not worry about the deletedCount check?

Comment (spiray) [reply src/api/faq/faq.controller.js]: I implemented that, tell me what you think.

Comment (ewenjo) [reply src/api/faq/faq.controller.js]: Both, add the length check before it tries to delete it. That would give a proper error message and avoid a unnecessary query to the database.

Comment (ewenjo) [reply src/api/faq/faq.controller.js]:
```
if (_id.length < 12) throw new ReferenceError('ID must be at least 12 characters..');
const deletedFAQ = await FAQ.deleteOne({ _id });
if (deletedFAQ.deletedCount === 0) throw new ReferenceError('There is no FAQ to delete with that ID.');
```

Comment (ewenjo) [reply src/api/faq/faq.controller.js]: You could also add the length check when trying to fetch a FAQ by ID

Comment (spiray) [reply src/api/faq/faq.controller.js]: Done!

**Comment by ewenjo** [comment]
Good work! Let's wait until #37 is merged, so you can change things accordingly before we merge.
You will have to change the error handler.

**Comment by ewenjo** [comment]
I did some research, this would be a better way to validate the IDs. 
`if (!/^[a-fA-F0-9]{24}$/.test(_id)) throw new ReferenceError('Invalid FAQ ID.');`

**Comment by spiray** [comment]
Added cross-env in front of test script and merged with latest PR. Will work on the error change a little later.

**Comment by spiray** [comment]
@ewenjo I think I am done with changing the error handling. Can you do one last review?

File: src/api/faq/faq.controller.js
Comment (ewenjo) [inline src/api/faq/faq.controller.js]: Should we include the ID of the created FAQ in the response?

File: package.json
Comment (ewenjo) [inline package.json]: Since the database connection isn't closed, I get this error after running the tests.
`Have you considered using "--detectOpenHandles" to detect async operations that kept running after all tests finished?`

Should we add "detectOpenHandles" to the test, close database connection after tests have been run, or just ignore the warning message?

**Review by ewenjo — APPROVED** [review]
lgtm, but take a look at the comments.

File: src/api/faq/faq.controller.js
Comment (spiray) [reply src/api/faq/faq.controller.js]: Done! Is that what you had in mind or should it be its own property?

File: package.json
Comment (spiray) [reply package.json]: I added it, I think we are fine with using detectOpenHandles.

File: src/api/faq/faq.controller.js
Comment (ewenjo) [reply src/api/faq/faq.controller.js]: That's what I had in mind, perfect the way it is now.

**Review by rabbitwerks — APPROVED** [review]
lgtm

**Pull request merged**
